### ISSUE TITLE
Why the nec_linear output loss while output prob for testing?

### ISSUE THREAD

**Issue by universewill** [Stonesjtu/Pytorch-NCE#14]
i don't understand the code below in sample.py:
```
# training mode
loss = nce_linear(target, input).mean()
print(loss.item())

# evaluation mode for fast probability computation
nce_linear.eval()
prob = nce_linear(target, input).mean()
print(prob.item())
```

Besides, why need target input parameter for inference?

**Comment by Stonesjtu** [comment]
If you train a model in NCE, it is already self-normalized. Only the logits of the target are needed to get the log-probability of target index.
